bukkit/src/main/java/com/dumbdogdiner/stickyapi/bukkit/util/StartupUtil.java

```diff
@@ -62,7 +63,15 @@ public static LocaleProvider setupLocale(@NotNull JavaPlugin plugin, @Nullable L
                     .severe("Failed to configure default locale file - perhaps you deleted it? Will create a new one.");
             // FIXME: This is horrible and needs to be improved
             try {
-                localeProvider.writeLocaleStream(plugin.getResource("messages.en_us.yml"), "messages.en_us.yml", true);
+                // Step 1:Load the yml from plugin.getResource (the internal .jar resource)
+                YamlProvider embeddedMessagesResource = new YamlProvider(plugin.getResource("messages.en_us.yml"));
```

Comment:
Maybe do an explicit check to see if this exists before copying it? Altho it is kinda covered by the `e.printStackTrace()` below...